src/main/java/com/microsoft/azure/springcloudplayground/controller/MainController.java

```diff
@@ -151,11 +136,16 @@ public ResponseEntity pushToGithub(@RequestBody @Nonnull ProjectRequest request,
             String username = token.getPrincipal().getAttributes().get("login").toString();
             GithubOperator operator = new GithubOperator(username, getAccessToken().getTokenValue());
             File dir = this.projectGenerator.generate(request);
+            triggerGenerateEvent(request.getMicroServices());
 
             try {
                 operator.createRepository(dir, request.getRepoName());
+                triggerGithubPushEvent(username, true);
+
                 return new ResponseEntity(HttpStatus.CREATED);
             } catch (GithubProcessException e) {
+                triggerGithubPushEvent(username, false);
```

Comment:
Is it possible to log details of exception?

Reply:
Yes, I wrapper all exception into GithubProcessException, and we can track the log details.